tidy3d/plugins/design/method.py

```diff
@@ -23,335 +20,780 @@ class Method(Tidy3dBaseModel, ABC):
-            raise ValueError(
-                f"Found different number of points: {num_points_each_dim} along each dimension. "
-                "This suggests a bug in the parameter sweep tool. "
-                "Please raise an issue on the front end GitHub repository."
-            )
-        return num_points_each_dim[0]
+        if all(isinstance(val, (list, Tuple)) for val in output):
+            if all(isinstance(val[0], (float, int)) for val in output):
+                float_out = []
+                aux_out = []
+                for val in output:
+                    if len(val) > 2:
+                        raise ValueError(
+                            "Unrecognized output from supplied post function. Too many elements in the return object, it should be a float and a list/tuple/dict."
+                        )
 
-    def _assemble_args(self, parameters: Tuple[ParameterType, ...]) -> Tuple[dict, int]:
-        """Sample design parameters, check the args are hashable and compute number of points."""
+                    float_out.append(val[0])
+                    aux_out.append(val[1])
 
-        fn_args = self.sample(parameters)
-        self.assert_hashable(fn_args)
-        num_points = self.get_num_points(fn_args)
-        return fn_args, num_points
+                # Float with aux_out
+                return (float_out, aux_out)
 
-    def run(self, parameters: Tuple[ParameterType, ...], fn: Callable) -> Tuple[Any]:
-        """Defines the search algorithm (sequential)."""
+            else:
+                raise ValueError(
+                    "Unrecognized output from supplied post function. The first element in the iterable object should be a float."
+                )
 
-        # get all function inputs
-        fn_args, num_points = self._assemble_args(parameters)
+        else:
+            raise ValueError(
+                "Unrecognized output from supplied post function. Output should be a float or an iterable object."
+            )
 
-        # for each point, construct the function inputs, run it, record output
-        result = []
-        for i in range(num_points):
-            fn_kwargs = {key: vals[i] for key, vals in fn_args.items()}
-            fn_output = fn(**fn_kwargs)
-            result.append(fn_output)
+    @staticmethod
+    def _flatten_and_append(list_of_lists: list[list], append_target: list) -> None:
+        """Flatten a list of lists and append the sublist to a new list."""
+        if list_of_lists is not None:
+            for sub_list in list_of_lists:
+                append_target.append(sub_list)
 
-        return fn_args, result
 
-    def _run_batch(
-        self, simulations: Dict[str, Simulation], path_dir: str = None, **kwargs
-    ) -> BatchData:
-        """Create a batch of simulations and run it. Mainly separated out for ease of testing."""
-        batch = web.Batch(simulations=simulations, simulation_type="tidy3d_design", **kwargs)
+class MethodSample(Method, ABC):
```

Comment:
I thought of this when you mentioned the issues with Batch and large number of tasks. Should we maybe provide a `batch_size : pd.PositiveInt = None` parameter? if `None` it does all of the tasks in a single batch. Otherwise it batches them? might be what's needed to handle super large amounts of sampling points